src/connectors/refinement-list/connectRefinementList.js

```diff
@@ -375,13 +388,14 @@ export default function connectRefinementList(renderFn, unmountFn = noop) {
       dispose({ state }) {
         unmountFn();
 
+        let newState;
```

Comment:
We can avoid the assignment with the call to `setQueryParameter` first, then we early return inside the branch.